### PR TITLE
[dua] handle AddressError

### PR DESCRIPTION
**Pull request by simonlingoogle**
This PR enhances DUA feature to handle `Address Error` by increasing the DAD counter and generating new DUA address if a DUA conflict is detected. 
- [x] Add test of DUA handling `/a/ae` using #5537

### CODE REVIEW

**Review by librasungirl — APPROVED** [review]

**Review by librasungirl — APPROVED** [review]

**Comment by simonlingoogle** [comment]
This PR is ready. But we need to merge https://github.com/openthread/openthread/pull/5537 first. @jwhui @librasungirl

File: tests/scripts/thread-cert/v1_2_test_dua_handle_address_error.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+        self._verify_dua_handle_address_error(ROUTER)
+        self._verify_dua_handle_address_error(FED)
+        self._verify_dua_handle_address_error(MED, is_med=True)
+
+    def _verify_dua_handle_address_error(self, nodeid, is_med=False):
+        dua = self.nodes[nodeid].get_addr(config.DOMAIN_PREFIX)
+        self.assertIsNotNone(dua)
+
+        # Ping the DUA to verify reachablity, and also fill the eidrloccache on BBR_1
```
Comment (librasungirl) [inline tests/scripts/thread-cert/v1_2_test_dua_handle_address_error.py:130]:
```suggestion
        # Ping the DUA to verify reachability, and also fill the EID cache on BBR_1
```

File: tests/scripts/thread-cert/v1_2_test_dua_handle_address_error.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        self.assertTrue(self.nodes[BBR_1].ping(dua))
+
+        self.simulator.go(WAIT_REDUNDANCE)
+
+        # Send fake /a/an to BBR_1 for ROUTER_1 DUA
```
Comment (librasungirl) [inline tests/scripts/thread-cert/v1_2_test_dua_handle_address_error.py:135]:
```suggestion
        # Send fake /a/an from ROUTER_1 to BBR_1 for nodeid's DUA
```

**Comment by simonlingoogle** [comment]
Labeled WIP to wait for #5537

**Review by jwhui — APPROVED** [review]

**Pull request merged**